# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: accessible/generic/LocalAccessible.cpp
# Commit: 2687a2f6d5d3
# Full Hash: 2687a2f6d5d355bfad12fb1746c92a68cf740e3f
# Author: Morgan Rae Reschenberg <mreschenberg@mozilla.com>
# Date: 2023-06-22 09:14:13
# Regressor Bug: 1837024
# File Overlap Count: 1
# Description:
#   Bug 1837024: Only do fuzzy hittesting on descendants of clipped accs r=Jamie
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D181166
# ==============================================================================

diff -r 54f3fddcb9bf -r 2687a2f6d5d3 accessible/generic/LocalAccessible.cpp
--- a/accessible/generic/LocalAccessible.cpp	Wed Jun 21 21:51:37 2023 +0000
+++ b/accessible/generic/LocalAccessible.cpp	Wed Jun 21 22:42:07 2023 +0000
@@ -3287,6 +3287,7 @@
   }
 
   bool boundsChanged = false;
+  nsIFrame* frame = GetFrame();
   if (aCacheDomain & CacheDomain::Bounds) {
     nsRect newBoundsRect = ParentRelativeBounds();
 
@@ -3336,6 +3337,12 @@
 
       fields->SetAttribute(nsGkAtoms::relativeBounds, std::move(boundsArray));
     }
+
+    if (frame && frame->ScrollableOverflowRect().IsEmpty()) {
+      fields->SetAttribute(nsGkAtoms::clip_rule, true);
+    } else if (aUpdateType != CacheUpdateType::Initial) {
+      fields->SetAttribute(nsGkAtoms::clip_rule, DeleteEntry());
+    }
   }
 
   if (aCacheDomain & CacheDomain::Text) {
@@ -3369,7 +3376,6 @@
     }
   }
 
-  nsIFrame* frame = GetFrame();
   if (aCacheDomain & (CacheDomain::Text | CacheDomain::Bounds) &&
       !HasChildren()) {
     // We cache line start offsets for both text and non-text leaf Accessibles